src/main/flight/feedforward.c

```diff
@@ -59,136 +64,144 @@ void feedforwardInit(const pidProfile_t *pidProfile)
     }
 }
 
-FAST_CODE_NOINLINE float feedforwardApply(int axis, bool newRcFrame, feedforwardAveraging_t feedforwardAveraging)
+FAST_CODE_NOINLINE float feedforwardApply(int axis, bool newRcFrame, feedforwardAveraging_t feedforwardAveraging, const float setpoint)
```

Comment:
Could this function be split into a calculate function and a get function?
Previously it was possible to just get the output by setting newRcFrame to false (which is still ugly, but possible), but now getting the latest FF factor requires knowing the setpoint and possibly altering the result.

Reply:
In the past, PID loop feedforward was only calculated on the acro rate setpoint derivative.  The `getRawSetpoint` value was used to get the raw setpoint value when a new RC Data packet arrived, and the output FF values were smoothed (upsampled) to the PID loop rate.

In angle mode, the setpoint value derived from angle mode is generated and updated in smoothed format every PID loop.  So the feedforward element needs to be calculated every PID loop, not when a new RC Frame appears, and since the duplicate detection code is already done in pid.c, the maths required will be different, and simpler.

It is possible to do all the feedforward calculations for angle mode in pid.c, but the approach here does the feedforward calculation for both angle mode and gyro mode in feedforward.c, with the selection depending on flight_mode.  

Note that angle_mode is automatically active in GPS Rescue, but not active in Horizon mode.

I don't see anywhere in feedforward.c that we change 'setpoint'.

I'm not sure if the above answers your question?